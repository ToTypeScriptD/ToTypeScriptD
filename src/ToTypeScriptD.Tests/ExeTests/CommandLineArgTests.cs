using ApprovalTests;
using System.IO;
using Xunit;

namespace ToTypeScriptD.Tests.ExeTests
{
    public class CommandLineArgTests : ExeTestBase
    {
        [Fact]
        public void ExeShouldGenerateHelpOnEmptyInput()
        {
            var result = Execute("").StdOut.StripVersionFromOutput();
            Approvals.Verify(result);
        }

        [Fact]
        public void ExeShouldGenerateHelpWithHelpArgs()
        {
            var result = Execute("--help").StdOut.StripVersionFromOutput();
            Approvals.Verify(result);
        }

        [Fact]
        public void ExeShouldGenerateHelpForDotNet()
        {
            var result = Execute(Options.DotNetCommandName).StdOut.StripVersionFromOutput();
            Approvals.Verify(result);
        }

        [Fact]
        public void ExeShouldGenerateHelpForWinMD()
        {
            var result = Execute(Options.WinmdCommandName).StdOut.StripVersionFromOutput();
            Approvals.Verify(result);
        }

        [Fact]
        public void ExeShouldGenerateHelpForDotNetWithBadInput()
        {
            var result = Execute(Options.DotNetCommandName + " --nogohere").StdOut.StripVersionFromOutput();
            Approvals.Verify(result);
        }

        [Fact]
        public void ExeShouldGenerateHelpForWinMDWithBadInput()
        {
            var result = Execute(Options.WinmdCommandName + " --nogohere").StdOut.StripVersionFromOutput();
            Approvals.Verify(result);
        }

        [Fact]
        public void ExeShouldGenerateOutputForMultipleWinmdFiles()
        {
            var result = Execute(@"winmd C:\Windows\System32\WinMetadata\Windows.Foundation.winmd C:\Windows\System32\WinMetadata\Windows.Networking.winmd");

            (result.StdOut.Length > 100).ShouldBeTrue(result.StdOut.Length + " should be greater than 100");
        }

        [Fact]
        public void VerifyExeGeneratedOutputForMultipleWinmdFiles()
        {
            var result = Execute(@"winmd C:\Windows\System32\WinMetadata\Windows.Foundation.winmd C:\Windows\System32\WinMetadata\Windows.System.winmd");

            result.StdOut.Verify();
        }

        [Fact]
        [ApprovalTests.Reporters.UseReporter(typeof(ApprovalTests.Reporters.P4MergeReporter))]
        public void ExeDuplicateAssemblyShouldStillOnlyGenerateOne()
        {
            var resultDup = Execute(@"winmd C:\Windows\System32\WinMetadata\Windows.Foundation.winmd C:\Windows\System32\WinMetadata\Windows.Foundation.winmd");
            var resultNonDup = Execute(@"winmd C:\Windows\System32\WinMetadata\Windows.Foundation.winmd");

            resultNonDup.ToString().StripHeaderGarbageromOutput()
                .DiffWith(resultDup.ToString().StripHeaderGarbageromOutput());
        }

        [Fact]
        public void ExeShouldBeAbleToGenerateSpecialTypes()
        {
            var resultDup = Execute("winmd --specialTypes");

            resultDup.Verify();
        }


        [Fact]
        public void ExeShouldGenerateWith2SpaceIndentation()
        {
            var resultDup = Execute("winmd --specialTypes ToTypeScriptD.Native.winmd --indentWith SpaceX2");

            resultDup.Verify();
        }

        [Fact]
        public void ExeShouldGiveHelpfulErrorWhenFilesNotFoundInUnknownDirectory()
        {
            var resultDup = Execute(@"winmd C:\TypeScriptD\TypeScriptD\TypeScriptD\Foo.dll C:\TypeScriptD\TypeScriptD\TypeScriptD\Foo.dll");

            resultDup.Verify();
        }

        [Fact]
        public void ExeShouldGiveHelpfulErrorWhenFilesNotFound()
        {
            var resultDup = Execute(@"winmd C:\TypeScriptD_FileNotFound_ThisShouldNotExistOnYourSystem.dll");

            resultDup.Verify();
        }

        [Fact]
        public void ExeShouldApplyRegexFilterOnTypes()
        {
            var resultDup = Execute(@"DotNet ..\..\..\bin\ToTypeScriptD.TestAssembly.CSharp.dll --regexFilter 'ToTypeScriptD.TestAssembly.CSharp.NamespaceSample|CSharp.NamespaceSample'");

            resultDup.Verify();
        }

    }
}
