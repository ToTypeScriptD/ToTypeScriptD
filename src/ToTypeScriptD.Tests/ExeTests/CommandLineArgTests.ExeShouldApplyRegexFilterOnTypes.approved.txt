//****************************************************************
//  Generated by:  ToTypeScriptD
//  Website:       http://github.com/ToTypeScriptD/ToTypeScriptD
//  Version:       v0.0.0000.0000 SHA1:0000000 - Debug
//  Date:          mm/dd/YYYY H:MM:SS PM
//
//  Assemblies:
//    ToTypeScriptD.TestAssembly.CSharp.dll
//
//****************************************************************



declare module ToTypeScriptD.TestAssembly.CSharp.NamespaceSample.Sample.SubSample {

    export interface TestClassB {
    }

}
declare module ToTypeScriptD.TestAssembly.CSharp.NamespaceSample.Sample {

    export interface TestClassA {
    }

}
declare module ToTypeScriptD.TestAssembly.CSharp.NamespaceSample.Sample2 {

    export interface TestClassC {
    }

}
declare module ToTypeScriptD.TestAssembly.CSharp.NamespaceSample {

    export interface TestClass {
    }

}

